azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -179,7 +183,15 @@ def put_vm_status(self, status_blob, sas_url, config_blob_type=None):
             if not HostPluginProtocol.is_default_channel():
                 logger.verbose("HostGAPlugin: Setting host plugin as default channel")
                 HostPluginProtocol.set_default_channel(True)
+
         except Exception as e:
+            # If the HostPlugin rejects the request,
+            # let the error continue, but set to use the HostPlugin
```

Comment:
I plead ignorance.  I don't understand this.  If HostGAPlugin rejects the request, why do you continue to use it?  Did your use of HostGAPlugin get reset in some way that I don't see?